extensions/wikia/PortableInfobox/PortableInfobox.setup.php

```diff
@@ -18,6 +18,22 @@
 $wgAutoloadClasses[ 'InfoboxServiceConnector' ] = $dir . 'InfoboxServiceConnector.class.php';
 $wgAutoloadClasses[ 'PortableInfoboxRenderService' ] = $dir . 'services/PortableInfoboxRenderService.class.php';
 
+// parser
+$wgAutoloadClasses[ 'PortableInfoboxMarkupParserService' ] = $dir . 'services/PortableInfoboxMarkupParserService.class.php';
+
+// autoloads values in the Wikia\PortableInfobox namespace
+spl_autoload_register( function( $class ) {
+	if ( substr_count( $class, 'Wikia\\PortableInfobox\\' ) > 0 ) {
+		$class = preg_replace( '/\\\\?Wikia\\\\PortableInfobox\\\\/', '', $class );
+		$file = __DIR__ . '/services/'.strtr( $class, '\\', '/' ).'.php';
+		if ( file_exists( $file ) ) {
+			require_once( $file );
```

Comment:
It's possible to use wgAutloadClasses, but not convinient (lots of small classes, that needs to be added manually)